aries_cloudagent/resolver/did_resolver.py

```diff
@@ -44,18 +44,20 @@ async def resolve(self, profile: Profile, did: Union[str, DID]) -> DIDDocument:
 
         raise DIDNotFound(f"DID {did} could not be resolved")
 
-    def _match_did_to_resolver(self, py_did: DID) -> BaseDIDResolver:
+    async def _match_did_to_resolver(
+        self, profile: Profile, py_did: DID
+    ) -> Sequence[BaseDIDResolver]:
         """Generate supported DID Resolvers.
 
         Native resolvers are yielded first, in registered order followed by
         non-native resolvers in registered order.
         """
-        valid_resolvers = list(
-            filter(
-                lambda resolver: resolver.supports(py_did),
-                self.did_resolver_registry.resolvers,
-            )
-        )
+        valid_resolvers = [
+            resolver
+            for resolver in self.did_resolver_registry.resolvers
+            if (await resolver.supports(profile, py_did), resolver)
```

Comment:
This doesn't look right. I think `bool(tuple)` will always evaluate to true unless it's an empty tuple.

Reply:
that it is, I will correct.
please comment on pr #84